Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -0,0 +1,1230 @@
+            }
+
+            BoxCollider boxCollider = targetObject.AddComponent<BoxCollider>();
+            bounds = boxCollider.bounds;
+            Destroy(boxCollider);
+            boundsMethod = BoundsCalculationMethod.Collider;
+            return bounds;
+        }
+        private Bounds GetSingleObjectBounds(GameObject gameObject)
+        {
+            Bounds bounds = new Bounds(Vector3.zero, Vector3.zero);
+            Component[] components = gameObject.GetComponents<Component>();
+            if (components.Length < 3)
+            {
+                return bounds;
+            }
+            BoxCollider boxCollider;
+            boxCollider = gameObject.GetComponent<BoxCollider>();
+            if (boxCollider == null)
+            {
+                boxCollider = gameObject.AddComponent<BoxCollider>();
+                bounds = boxCollider.bounds;
+                DestroyImmediate(boxCollider);
+            }
+            else
+            {
+                bounds = boxCollider.bounds;
+            }
+
+            return bounds;
+        }
+        private void SetMaterials()
+        {
+            if (wireframeMaterial == null)
+            {
+                float[] color = { 1.0f, 1.0f, 1.0f, 0.75f };
+
+                Shader.EnableKeyword("_InnerGlow");
+                Shader shader = Shader.Find("Mixed Reality Toolkit/Standard");
+
+                wireframeMaterial = new Material(shader);
+                wireframeMaterial.SetColor("_Color", new Color(0.0f, 0.63f, 1.0f));
+            }
+            if (handleMaterial == null && handleMaterial != wireframeMaterial)
+            {
+                float[] color = { 1.0f, 1.0f, 1.0f, 0.75f };
+
+                Shader.EnableKeyword("_InnerGlow");
+                Shader shader = Shader.Find("Mixed Reality Toolkit/Standard");
+
+                handleMaterial = new Material(shader);
+                handleMaterial.SetColor("_Color", new Color(0.0f, 0.63f, 1.0f));
+                handleMaterial.SetFloat("_InnerGlow", 1.0f);
+                handleMaterial.SetFloatArray("_InnerGlowColor", color);
+            }
+            if (handleGrabbedMaterial == null && handleGrabbedMaterial != handleMaterial && handleGrabbedMaterial != wireframeMaterial)
+            {
+                float[] color = { 1.0f, 1.0f, 1.0f, 0.75f };
+
+                Shader.EnableKeyword("_InnerGlow");
+                Shader shader = Shader.Find("Mixed Reality Toolkit/Standard");
+
+                handleGrabbedMaterial = new Material(shader);
+                handleGrabbedMaterial.SetColor("_Color", new Color(0.0f, 0.63f, 1.0f));
+                handleGrabbedMaterial.SetFloat("_InnerGlow", 1.0f);
+                handleGrabbedMaterial.SetFloatArray("_InnerGlowColor", color);
+            }
+        }
+        private void InitializeDataStructures()
+        {
+            rigRoot = new GameObject("rigRoot");
+            rigRoot.hideFlags = HideFlags.HideInHierarchy | HideFlags.HideInInspector;
```

Comment:
I'm not sure we should be hiding objects in the scene.

Reply:
As this is a generated object, I think it's okay.

Reply:
the handles shouldn't show up as a long stream of cubes in the hierarchy